sys-cluster/lmod/lmod-8.7.55.ebuild

```diff
@@ -0,0 +1,136 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-{1..3} )
+inherit autotools lua-single prefix
+
+DESCRIPTION="Environment Module System based on Lua"
+HOMEPAGE="https://lmod.readthedocs.io/en/latest https://github.com/TACC/Lmod"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/TACC/Lmod"
+else
+	SRC_URI="https://github.com/TACC/Lmod/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	S="${WORKDIR}"/Lmod-${PV}
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="+auto-swap +cache duplicate-paths test"
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+RDEPEND="${LUA_DEPS}
+	dev-lang/tcl
+	dev-lang/tk
+	$(lua_gen_cond_dep '
+		>=dev-lua/luafilesystem-1.8.0[${LUA_USEDEP}]
+		dev-lua/luajson[${LUA_USEDEP}]
+		dev-lua/luaposix[${LUA_USEDEP}]
+		dev-lua/lua-term[${LUA_USEDEP}]
+	')
+	virtual/pkgconfig
+"
+DEPEND="${RDEPEND}"
+BDEPEND="${RDEPEND}
```

Comment:
Is this correct? All runtime dependencies are executed during the build?

Reply:
`bc` is called during the build, so it definitely needs to be there.

For the Lua modules: the `configure` script may run a Lua script (https://github.com/TACC/Lmod/blob/8.7.55/proj_mgmt/luaModuleAvailable) that tries to import the required Lua modules (and may fail if it cannot find them), and the tests also need/use them. Having read https://devmanual.gentoo.org/general-concepts/dependencies/#dependency-types, it's not entirely clear to me where they belong. Could you give some advice if they should be in DEPEND or BDPEND (or even in the conditional test part of BDEPEND?) in this case?

Reply:
> Could you give some advice if they should be in DEPEND or BDPEND (or even in the conditional test part of BDEPEND?) in this case?

I have never used lua myself, but my understanding is that it is a scripting language. BDEPEND versus DEPEND becomes a bit ambiguous in those cases, the convention used for python packages in cases where e.g. `setup.py` imports some other python package is to put those packages in BDEPEND. So I suggest we do the same here, in that case `BDEPEND="${RDEPEND}` would be correct and `DEPEND="${RDEPEND}"` can be removed.

With regards to test dependencies, portage implicitly pulls in the RDEPENDs as BDEPENDs when FEATURES=test, so `BDEPEND="test? ( ${RDEPEND} )"` is never required.

Reply:
Thanks! I've removed `DEPEND="${RDEPEND}"` (and it worked fine for me).